### ISSUE TITLE
Multi-schema databases

### ISSUE THREAD

**Issue by joskuijpers** [theory/sqitch#239]
Hi,

I find this piece of software awesome, and it is just what I was looking for.
However, I have one question.

I am making a piece of SaaS software, where each tenant has its own schema. In your presentations and  examples you always work all the sql code on a single schema.

Is it (easily) possible to make the changes apply to any N schemas in my database? Maybe some way to specify the default schema in the command line?
This is only needed in production mode. Development will just be on one schema.

With kind regards,

Jos

**Comment by theory** [comment]
Which engine are you using? We do this a lot in our PostgreSQL projects. I just put this at the top of each script:

    SET search_path TO config,public;

Where `config` is the schema I want to create objects in. Sometimes I just schema qualify, rather than set the search path, especially in revert scripts where I'm just dropping a single object:

    DROP TABLE config.users;

**Comment by joskuijpers** [comment]
I am using PG.

If I would be using 20 schemas (hypothetical) with all the same layout, would I need to add that line to every file, and change it 20 times for a single update? :s I think you understood me wrong :p

Thanks.

**Comment by theory** [comment]
In that case, put that line in every file but use a psql variable, like this:

    SET search_path TO :"schema",public;

Then deploy to specific schemas like this:

    sqitch deploy --set schema=foo
    sqitch deploy --set schema=bar

Also, set the default value in the config in case someone forgets to set it on the command-line:

    sqitch config deploy.variables.schema baked

**Comment by joskuijpers** [comment]
Thanks, this looks awesome. Then all I need is a tiny script to run this for every schema. Awesome, I will try this out.

Thank you for the fast responses. Both Sqitch and PGtap are resolutions to my problems :)

**Comment by theory** [comment]
Great, glad to hear it.

**Comment by joskuijpers** [comment]
@theory I hoped this would work, but I tried it now and it does not.
I think it has to do with that the sqitch schema does not incorporate my variables:

deploy/schema.sql
```sql
BEGIN;

CREATE SCHEMA :"schema";

COMMIT;
```

Then I run:
```sh
sqitch deploy db:pg://<theaddress>/postgres --set schema=test
```

That works and creates the test schema.
Now, I want to run it again but with another schema. So I replace test with test2, and I want it to run it all again, but with test2 as schema.
It says nothing to deploy (up-to-date).

I know why this is, but how to solve it? Am I forced by sqitch to work with a database-per-tenant?
